app/Models/CopyRepository.php

```
@@ -89,6 +138,81 @@ public function create(int $bookId, string $numeroInventario, string $stato = 'd
+     * the wrong end (Duplicate entry '…-C2'). Here N walks up from 1, filling any
+     * gap left by a removed copy, and every candidate is checked against BOTH the
+     * codes generated in this batch and the whole `copie` table (global unique key),
+     * so the returned codes can never duplicate an existing one. The "-C{N}" suffix
+     * is applied uniformly (never a bare base), keeping the naming consistent.
+     *
+     * @return list<string>
+     */
+    public function allocateInventoryCodes(string $base, int $howMany): array
+    {
+        $codes = [];
+        $n = 1;
```

PHPMD (2.15.0)

[warning] Avoid variables with short names like $n. Configured minimum length is 3. (undefined)

(ShortVariable)